Update CHANGELOG.md to reflect 5.9 scope (#6839)
apple/swift-package-manager: CHANGELOG.md

Swift Next
-----------

* [#5728]

  In packages that specify resources using a future tools version, the generated resource bundle accessor will import `Foundation.Bundle` for its own implementation only. _Clients_ of such packages therefore no longer silently import `Foundation`, preventing inadvertent use of Foundation extensions to standard library APIs, which helps to avoid unexpected code size increases.

Swift 5.9
-----------

* [SE-0386]

  SwiftPM packages can now use `package` as a new access modifier, allowing accessing symbols in another target / module within the same package without making it public.

* [SE-0391]

>   SwiftPM can now publish to a registry following the pulish spec defined in [SE-0391]

Author: ```suggestion
  SwiftPM can now publish to a registry following the publishing spec as defined in [SE-0391]
```

>   SwiftPM can now publish to a registry following the publishing spec as defined in [SE-0391]

Reviewer: ```suggestion
  SwiftPM can now publish to a registry following the publishing spec as defined in [SE-0391]. SwiftPM also gains support for signed packages. Trust-on-first-use (TOFU) check which includes only fingerprints (e.g., checksums) previously has been extended to include signing identities, and it is enforced for source archives as well as package manifests.
```

* [SE-0391]

  SwiftPM can now publish to a registry following the publishing spec as defined in [SE-0391]. SwiftPM also gains support for signed packages. Trust-on-first-use (TOFU) check which includes only fingerprints (e.g., checksums) previously has been extended to include signing identities, and it is enforced for source archives as well as package manifests.

* [#5966]

  Plugin compilation can be influenced by using `-Xbuild-tools-swiftc` arguments in the SwiftPM command line. This is similar to the existing mechanism for influencing the manifest compilation using `-Xmanifest` arguments. Manifest compilation will also be influenced by `-Xbuild-tools-swiftc`, but only if no other `-Xmanifest` arguments are provided. Using `-Xmanifest` will show a deprecation message. `-Xmanifest` will be removed in the future.

* [#6060]

  Support for building plugin dependencies for the host when cross-compiling.

* [#6067]

  Basic support for a new `.embedInCode` resource rule which allows embedding the contents of the resource into the executable code by generating a byte array, e.g.

  ```
  struct PackageResources {

>     static let best_txt: [UInt8] = [104,101,108,108,111,32,119,111,114,108,100,10]

Reviewer 2: Hm, why is a space here? It definitely won't be there in the generated code :)

Author: just to improve formatting?

  }
  ```

* [#6111]

  Package creation using `package init` now also supports the build tool plugin and command plugin types.

* [#6114]

  Added a new `allowNetworkConnections(scope:reason:)` for giving a command plugin permissions to access the network. Permissions can be scoped to Unix domain sockets in general or specifically for Docker, as well as local or remote IP connections which can be limited by port. For non-interactive use cases, there is also a `--allow-network-connections` commandline flag to allow network connections for a particular scope.

* [#6144]

  Remove the `system-module` and `manifest` templates and clean up the remaining `empty`, `library`, and `executable` templates so they include the minimum information needed to get started, with links to documentation in the generated library, executable, and test content.

* [#6185], [#6200]

  Add a new `CompilerPluginSupport` module which contains the definition for macro targets. Macro targets allow authoring and distribution of custom Swift macros such as [expression macros](https://github.com/apple/swift-evolution/blob/main/proposals/0382-expression-macros.md).

* [#6276]

>   Add newbuild setting in the package manifest that enables Swift/C++ Interoperability for a given Swift target.

Reviewer 3: One more typo nit

```suggestion
  Add new build setting in the package manifest that enables Swift/C++ Interoperability for a given Swift target.
```

* [#6144]

  Remove the `system-module` and `manifest` templates and clean up the remaining `empty`, `library`, and `executable` templates so they include the minimum information needed to get started, with links to documentation in the generated library, executable, and test content.

* [#6185], [#6200]

  Add a new `CompilerPluginSupport` module which contains the definition for macro targets. Macro targets allow authoring and distribution of custom Swift macros such as [expression macros](https://github.com/apple/swift-evolution/blob/main/proposals/0382-expression-macros.md).

* [#6294]

  When a package contains a single target, sources may be distributed anywhere within the `./Sources` directory. If sources are placed in a subdirectory under `./Sources/<target>`, or there is more than one target, the existing expectation for sources apply.

* [#6540]

  Build tool plugins can be used with C-family targets

* [#6663]

>   Add `visionOS` as a platform alongside `iOS`, `tvOS`, `watchOS`, and the other

Reviewer 3: ```suggestion
  Add `visionOS` as a platform alongside `iOS`, `tvOS`, `watchOS`, and the other
  
* [#6023], [#6425], [#6425]

New `swift experimental-sdk` command is now available for managing Swift SDK bundles that follow the format described in [SE-0387]: "Swift SDKs for Cross-Compilation".
```

* [#6144]

  Remove the `system-module` and `manifest` templates and clean up the remaining `empty`, `library`, and `executable` templates so they include the minimum information needed to get started, with links to documentation in the generated library, executable, and test content.

* [#6185], [#6200]

  Add a new `CompilerPluginSupport` module which contains the definition for macro targets. Macro targets allow authoring and distribution of custom Swift macros such as [expression macros](https://github.com/apple/swift-evolution/blob/main/proposals/0382-expression-macros.md).

* [#6294]

  When a package contains a single target, sources may be distributed anywhere within the `./Sources` directory. If sources are placed in a subdirectory under `./Sources/<target>`, or there is more than one target, the existing expectation for sources apply.

* [#6540]

  Build tool plugins can be used with C-family targets

* [#6663]

  Add `visionOS` as a platform alongside `iOS`, `tvOS`, `watchOS`, and the other
  
* [#5911], [#5922], [#6023], [#6425]

> New `swift experimental-sdk` command is now available for managing Swift SDK bundles that follow the format described in [SE-0387]: "Swift SDKs for Cross-Compilation".

Reviewer 3: Formatting fix: lack of this indentation made it render differently.
```suggestion
  New `swift experimental-sdk` command is now available for managing Swift SDK bundles that follow the format described in [SE-0387]: "Swift SDKs for Cross-Compilation".
```

[SE-0339]: https://github.com/apple/swift-evolution/blob/main/proposals/0339-module-aliasing-for-disambiguation.md
[SE-0362]: https://github.com/apple/swift-evolution/blob/main/proposals/0362-piecemeal-future-features.md
[SE-0378]: https://github.com/apple/swift-evolution/blob/main/proposals/0378-package-registry-auth.md

> [SE-0386]: https://github.com/apple/swift-evolution/blob/main/proposals/0386-package-access-modifier.md

Reviewer 3: ```suggestion
[SE-0386]: https://github.com/apple/swift-evolution/blob/main/proposals/0386-package-access-modifier.md
[SE-0387]: https://github.com/apple/swift-evolution/blob/main/proposals/0387-cross-compilation-destinations.md
```
